File: back/Documento.py
class Documento:
    def __init__(self, db):
        self.db = db

    def add_documento(self, caminho_arquivo, nome_arquivo, id_projeto):
        print(f"Adicionando documento: caminho_arquivo={caminho_arquivo}, nome_arquivo={nome_arquivo}, id_projeto={id_projeto}")
        query = "INSERT INTO documento (Caminho_Arquivo, Nome_Arquivo, ID_Projeto) VALUES (%s, %s, %s)"
        try:
            self.db.cursor.execute(query, (caminho_arquivo, nome_arquivo, id_projeto))
            self.db.conn.commit()
            print(f"Documento adicionado com sucesso: {self.db.cursor.lastrowid}")
            return self.db.cursor.lastrowid
        except Exception as e:
            print(f"Erro ao adicionar documento: {e}")
            self.db.conn.rollback()
            raise

    def update_documento(self, documento_id, nome_arquivo, caminho_arquivo):
        print(f"Atualizando documento: id={documento_id}, caminho_arquivo={caminho_arquivo}, nome_arquivo={nome_arquivo}")
        query = "UPDATE documento SET Nome_Arquivo = %s, Caminho_Arquivo = %s WHERE ID = %s"
        try:
            self.db.cursor.execute(query, (nome_arquivo, caminho_arquivo, documento_id))
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao atualizar documento: {e}")
            self.db.conn.rollback()
            raise

    def delete_documento(self, documento_id):
        try:
            # Deletar as referências na tabela evidencia primeiro
            delete_evidencia_query = "DELETE FROM evidencia WHERE ID_Documento = %s"
            self.db.cursor.execute(delete_evidencia_query, (documento_id,))
            self.db.conn.commit()

            # Deletar o documento na tabela documento
            delete_documento_query = "DELETE FROM documento WHERE ID = %s"
            self.db.cursor.execute(delete_documento_query, (documento_id,))
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao deletar documento: {e}")
            self.db.conn.rollback()
            raise

    def get_documentos_by_projeto(self, id_projeto):
        query = "SELECT * FROM documento WHERE ID_Projeto = %s"
        self.db.cursor.execute(query, (id_projeto,))
        return self.db.cursor.fetchall()

File: backjs/classes/Documento.js
class Documento {
    constructor(db) {
        this.db = db;
    }

    async addDocumento(caminhoArquivo, nomeArquivo, idProjeto) {
        const query = `
            INSERT INTO documento (Caminho_Arquivo, Nome_Arquivo, ID_Projeto) 
            VALUES (?, ?, ?)
        `;
        try {
            const result = await this.db.executeQuery(query, [caminhoArquivo, nomeArquivo, idProjeto]);
            const documentoId = result.insertId;
            this.db.conn.commit();
            return documentoId;
        } catch (err) {
            console.error("Erro ao adicionar documento:", err);
            this.db.conn.rollback();
            throw err;
        }
    }

    async updateDocumento(documentoId, nomeArquivo, caminhoArquivo) {
        const query = `
            UPDATE documento 
            SET Nome_Arquivo = ?, Caminho_Arquivo = ? 
            WHERE ID = ?
        `;
        try {
            await this.db.executeQuery(query, [nomeArquivo, caminhoArquivo, documentoId]);
            this.db.conn.commit();
        } catch (err) {
            console.error("Erro ao atualizar documento:", err);
            this.db.conn.rollback();
            throw err;
        }
    }

    async deleteDocumento(documentoId) {
        try {
            // Deletar as referências na tabela 'evidencia'
            const deleteEvidenciaQuery = "DELETE FROM evidencia WHERE ID_Documento = ?";
            await this.db.executeQuery(deleteEvidenciaQuery, [documentoId]);

            // Deletar o documento na tabela 'documento'
            const deleteDocumentoQuery = "DELETE FROM documento WHERE ID = ?";
            await this.db.executeQuery(deleteDocumentoQuery, [documentoId]);

            this.db.conn.commit();
        } catch (err) {
            console.error("Erro ao deletar documento:", err);
            this.db.conn.rollback();
            throw err;
        }
    }

    async getDocumentosByProjeto(idProjeto) {
        const query = "SELECT * FROM documento WHERE ID_Projeto = ?";
        try {
            const documentos = await this.db.fetchAll(query, [idProjeto]);
            return documentos;
        } catch (err) {
            console.error("Erro ao buscar documentos por projeto:", err);
            throw err;
        }
    }
}

module.exports = Documento;